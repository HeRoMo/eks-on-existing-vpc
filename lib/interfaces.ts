export interface AvailabilityZones {
  [key: string]: { public: string; private: string }
}
