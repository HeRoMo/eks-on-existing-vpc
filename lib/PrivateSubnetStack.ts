import { Construct } from '@aws-cdk/core';
import { PrivateSubnet, ISubnet } from '@aws-cdk/aws-ec2';

import { AvailabilityZones } from './interfaces';
import BaseStack from './BaseStack';

export default class PrivateSubnetStack extends BaseStack {
  public readonly subnets: ISubnet[];
  constructor(
    scope: Construct,
    id: string,
    vpcId: string,
    natGwId: string,
    availabilityZones: AvailabilityZones) {
    super(scope, id);

    this.subnets = Object.entries(availabilityZones).map(([availabilityZone, { private: cidrBlock } ]) => {
      const subnet = new PrivateSubnet(this, `privateSubnet-${availabilityZone}`, {
        availabilityZone,
        cidrBlock,
        vpcId,
        mapPublicIpOnLaunch: true,
      });
      subnet.addDefaultNatRoute(natGwId);
      return subnet;
    });
  }
}
