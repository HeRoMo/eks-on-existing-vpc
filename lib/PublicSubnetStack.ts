import { Construct, ConcreteDependable } from '@aws-cdk/core';
import { PublicSubnet, ISubnet } from '@aws-cdk/aws-ec2';

import { AvailabilityZones } from './interfaces';
import BaseStack from './BaseStack';

export default class PublicSubnetStack extends BaseStack {
  public readonly subnets: ISubnet[];

  constructor(
    scope: Construct,
    id: string,
    vpcId: string,
    igwId: string,
    availabilityZones: AvailabilityZones) {
    super(scope, id);

    const subnets = Object.entries(availabilityZones).map(([availabilityZone, { public: cidrBlock }]) => {
      const subnet = new PublicSubnet(this, `publicSubnet-${availabilityZone}`, {
        availabilityZone,
        cidrBlock,
        vpcId,
        mapPublicIpOnLaunch: true,
      });
      subnet.addDefaultInternetRoute(igwId, new ConcreteDependable());
      return subnet;
    })
  }
}
