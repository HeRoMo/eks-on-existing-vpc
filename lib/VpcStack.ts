import { Construct } from '@aws-cdk/core';
import { Vpc, IVpc } from '@aws-cdk/aws-ec2';

import BaseStack from './BaseStack';

export default class VpcStack extends BaseStack {
  public readonly vpc: IVpc;

  constructor(scope: Construct, id: string, vpcId: string) {
    super(scope, id);

    this.vpc = Vpc.fromLookup(this, 'DefaultVpc', { vpcId });
  }
}
