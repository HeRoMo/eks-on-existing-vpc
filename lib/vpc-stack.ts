import { Construct, Stack, StackProps, ConcreteDependable } from '@aws-cdk/core';
import { Vpc, PrivateSubnet, PublicSubnet, ISubnet, IVpc } from '@aws-cdk/aws-ec2';

const availabilityZones = ['ap-northeast-1a', 'ap-northeast-1c', 'ap-northeast-1d'];

const VPC_ID = process.env.VPC_ID || '';
const IGW_ID = process.env.IGW_ID || '';
const NAT_GW_ID = process.env.NAT_GW_ID || '';

export class PrivateSubnetStack extends Stack {
  public readonly subnets: ISubnet[];
  constructor(scope: Construct, id: string, vpcId: string, props?: StackProps) {
    super(scope, id, props);

    let netId = 50;
    this.subnets = availabilityZones.map((availabilityZone) => {
      const subnet = new PrivateSubnet(this, `privateSubnet-${availabilityZone}`, {
        availabilityZone,
        cidrBlock: `172.31.${netId += 1}.0/24`,
        vpcId,
        mapPublicIpOnLaunch: true,
      });
      subnet.addDefaultNatRoute(NAT_GW_ID);
      return subnet;
    });
  }
}

export class PublicSubnetStack extends Stack {
  public readonly subnets: ISubnet[];

  constructor(scope: Construct, id: string, vpcId: string, props?: StackProps) {
    super(scope, id, props);

    let netId = 60;
    const subnets = availabilityZones.map((availabilityZone) => {
      const subnet = new PublicSubnet(this, `publicSubnet-${availabilityZone}`, {
        availabilityZone,
        cidrBlock: `172.31.${netId += 1}.0/24`,
        vpcId,
        mapPublicIpOnLaunch: true,
      });
      subnet.addDefaultInternetRoute(IGW_ID, new ConcreteDependable());
      return subnet;
    })
  }
}

export class VpcStack extends Stack {
  public readonly vpc: IVpc;

  constructor(scope: Construct, id: string, props?: StackProps) {
    super(scope, id, props);

    this.vpc = Vpc.fromLookup(this, 'DefaultVpc', {
      vpcId: VPC_ID,
    });
  }
}
