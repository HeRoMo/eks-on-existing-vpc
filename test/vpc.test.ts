import { expect as expectCDK, matchTemplate, MatchStyle, SynthUtils } from '@aws-cdk/assert';
import { App } from '@aws-cdk/core';
import { VpcStack, PrivateSubnetStack, PublicSubnetStack } from '../lib/vpc-stack';

const vpcId = process.env.VPC_ID || '';

test('VpcStack', () => {
    const app = new App();
    const stack = new VpcStack(app, 'MyTestStack');
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});

test('VpcStack', () => {
  const app = new App();

  const stack = new PrivateSubnetStack(app, 'MyTestStack', vpcId);
  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});

test('VpcStack', () => {
  const app = new App();
  const stack = new PublicSubnetStack(app, 'MyTestStack', vpcId);
  expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});
