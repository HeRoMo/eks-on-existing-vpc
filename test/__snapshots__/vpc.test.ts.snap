// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VpcStack 1`] = `Object {}`;

exports[`VpcStack 2`] = `
Object {
  "Resources": Object {
    "privateSubnetapnortheast1aDefaultRoute3AC536A6": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": "nat-99999999999999999",
        "RouteTableId": Object {
          "Ref": "privateSubnetapnortheast1aRouteTable69CBA704",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "privateSubnetapnortheast1aRouteTable69CBA704": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestStack/privateSubnet-ap-northeast-1a",
          },
        ],
        "VpcId": "vpc-99999999",
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "privateSubnetapnortheast1aRouteTableAssociationFE5AA7E4": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "privateSubnetapnortheast1aRouteTable69CBA704",
        },
        "SubnetId": Object {
          "Ref": "privateSubnetapnortheast1aSubnetD7D78EEF",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "privateSubnetapnortheast1aSubnetD7D78EEF": Object {
      "Properties": Object {
        "AvailabilityZone": "ap-northeast-1a",
        "CidrBlock": "172.31.51.0/24",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestStack/privateSubnet-ap-northeast-1a",
          },
        ],
        "VpcId": "vpc-99999999",
      },
      "Type": "AWS::EC2::Subnet",
    },
    "privateSubnetapnortheast1cDefaultRouteAEEC5E5B": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": "nat-99999999999999999",
        "RouteTableId": Object {
          "Ref": "privateSubnetapnortheast1cRouteTableE152E62E",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "privateSubnetapnortheast1cRouteTableAssociation3829C8F8": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "privateSubnetapnortheast1cRouteTableE152E62E",
        },
        "SubnetId": Object {
          "Ref": "privateSubnetapnortheast1cSubnetE57E57C3",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "privateSubnetapnortheast1cRouteTableE152E62E": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestStack/privateSubnet-ap-northeast-1c",
          },
        ],
        "VpcId": "vpc-99999999",
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "privateSubnetapnortheast1cSubnetE57E57C3": Object {
      "Properties": Object {
        "AvailabilityZone": "ap-northeast-1c",
        "CidrBlock": "172.31.52.0/24",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestStack/privateSubnet-ap-northeast-1c",
          },
        ],
        "VpcId": "vpc-99999999",
      },
      "Type": "AWS::EC2::Subnet",
    },
    "privateSubnetapnortheast1dDefaultRouteCBED6AF5": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "NatGatewayId": "nat-99999999999999999",
        "RouteTableId": Object {
          "Ref": "privateSubnetapnortheast1dRouteTable31BA8D9E",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "privateSubnetapnortheast1dRouteTable31BA8D9E": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestStack/privateSubnet-ap-northeast-1d",
          },
        ],
        "VpcId": "vpc-99999999",
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "privateSubnetapnortheast1dRouteTableAssociation1B33F8A6": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "privateSubnetapnortheast1dRouteTable31BA8D9E",
        },
        "SubnetId": Object {
          "Ref": "privateSubnetapnortheast1dSubnet596F574C",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "privateSubnetapnortheast1dSubnet596F574C": Object {
      "Properties": Object {
        "AvailabilityZone": "ap-northeast-1d",
        "CidrBlock": "172.31.53.0/24",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestStack/privateSubnet-ap-northeast-1d",
          },
        ],
        "VpcId": "vpc-99999999",
      },
      "Type": "AWS::EC2::Subnet",
    },
  },
}
`;

exports[`VpcStack 3`] = `
Object {
  "Resources": Object {
    "publicSubnetapnortheast1aDefaultRoute13BEB917": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": "igw-99999999",
        "RouteTableId": Object {
          "Ref": "publicSubnetapnortheast1aRouteTableAD7C168D",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "publicSubnetapnortheast1aRouteTableAD7C168D": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestStack/publicSubnet-ap-northeast-1a",
          },
        ],
        "VpcId": "vpc-99999999",
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "publicSubnetapnortheast1aRouteTableAssociationCD1D3738": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "publicSubnetapnortheast1aRouteTableAD7C168D",
        },
        "SubnetId": Object {
          "Ref": "publicSubnetapnortheast1aSubnet553AC7A4",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "publicSubnetapnortheast1aSubnet553AC7A4": Object {
      "Properties": Object {
        "AvailabilityZone": "ap-northeast-1a",
        "CidrBlock": "172.31.61.0/24",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestStack/publicSubnet-ap-northeast-1a",
          },
        ],
        "VpcId": "vpc-99999999",
      },
      "Type": "AWS::EC2::Subnet",
    },
    "publicSubnetapnortheast1cDefaultRoute14669087": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": "igw-99999999",
        "RouteTableId": Object {
          "Ref": "publicSubnetapnortheast1cRouteTableAEF2BBEE",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "publicSubnetapnortheast1cRouteTableAEF2BBEE": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestStack/publicSubnet-ap-northeast-1c",
          },
        ],
        "VpcId": "vpc-99999999",
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "publicSubnetapnortheast1cRouteTableAssociation7B6B56B2": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "publicSubnetapnortheast1cRouteTableAEF2BBEE",
        },
        "SubnetId": Object {
          "Ref": "publicSubnetapnortheast1cSubnetED1EF36F",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "publicSubnetapnortheast1cSubnetED1EF36F": Object {
      "Properties": Object {
        "AvailabilityZone": "ap-northeast-1c",
        "CidrBlock": "172.31.62.0/24",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestStack/publicSubnet-ap-northeast-1c",
          },
        ],
        "VpcId": "vpc-99999999",
      },
      "Type": "AWS::EC2::Subnet",
    },
    "publicSubnetapnortheast1dDefaultRouteAE72E3F6": Object {
      "Properties": Object {
        "DestinationCidrBlock": "0.0.0.0/0",
        "GatewayId": "igw-99999999",
        "RouteTableId": Object {
          "Ref": "publicSubnetapnortheast1dRouteTable5A11ADE7",
        },
      },
      "Type": "AWS::EC2::Route",
    },
    "publicSubnetapnortheast1dRouteTable5A11ADE7": Object {
      "Properties": Object {
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestStack/publicSubnet-ap-northeast-1d",
          },
        ],
        "VpcId": "vpc-99999999",
      },
      "Type": "AWS::EC2::RouteTable",
    },
    "publicSubnetapnortheast1dRouteTableAssociation70C40C25": Object {
      "Properties": Object {
        "RouteTableId": Object {
          "Ref": "publicSubnetapnortheast1dRouteTable5A11ADE7",
        },
        "SubnetId": Object {
          "Ref": "publicSubnetapnortheast1dSubnetD1C92021",
        },
      },
      "Type": "AWS::EC2::SubnetRouteTableAssociation",
    },
    "publicSubnetapnortheast1dSubnetD1C92021": Object {
      "Properties": Object {
        "AvailabilityZone": "ap-northeast-1d",
        "CidrBlock": "172.31.63.0/24",
        "MapPublicIpOnLaunch": true,
        "Tags": Array [
          Object {
            "Key": "Name",
            "Value": "MyTestStack/publicSubnet-ap-northeast-1d",
          },
        ],
        "VpcId": "vpc-99999999",
      },
      "Type": "AWS::EC2::Subnet",
    },
  },
}
`;
